microsoft/testsuites/mdatp/mdatp.py

```diff
@@ -22,6 +24,59 @@
 EXIT_ONBOARD_INFO_FOUND = 255
 
 
+def working_path_workaround(node: Node) -> None:
+    # EXAMPLE WORKAROUND:
+    # - Get the mount point for the path.
+    # - Check if the mount point is mounted noexec.
+    # - if yes, remount with exec flag
+    working_path = node.get_working_path()
+    mount_info = node.tools[Df].get_mount_info_for_dir(
+        directory=working_path.as_posix()
+    )
+    if not mount_info:
+        node.log.warn(
+            f"Could not locate mount info for directory "
+            f"{working_path.as_posix()}. "
+            "Test may fail due to noexec permissions error. "
+            "Or due to the working path not existing."
+        )
+    else:
+        if check_noexec_partition(node, mount_info):
+            node.log.info(
+                f"Working path in {mount_info.mountpoint} is mounted noexec!!"
+            )
+            fix_noexec_working_path(node, mount_info)
+
+
+def check_noexec_partition(node: Node, partition: PartitionInfo) -> bool:
+    # an example. I don't have time to implement this in mount,
+    # we currently discard the mount options so they're not exposed
+    # in tools.Mount
+    for part in node.tools[Mount].get_partition_info():
+        node.log.debug(f"checking partition {str(part)}...")
+        if part.mount_point == partition.mountpoint:
+            return "noexec" in part.options
+    # This path would mean that df and mount are returning different
+    # info about the mount points on the VM.
+    # There should be no way for this to happen, assert if it does
+    assert f"Could not find partition info for {partition.mountpoint}!"
```

Comment:
This assert doesn't work. It asserts a string is not None. So it won't be failed.

Reply:
Oh wow, yep! Thank you so much, I wanted to finish this example so I wrote this around 1AM. Will fix

Reply:
Hmm maybe this should also be a warning too and not an assert.

 I don't want to raise an exception in this code path since almost all images (>99%) don't need this workaround.
The problem also isn't a bug in this test, it's a LISA bug where we are not paying attention to mount options when creating our working paths. We're also not paying attention to file ACLs, but that's another issue. I don't have time or enough context about the inheritances in Node to fix the general case. I'm hoping we can create a bug and someone can hoist this code to the appropriate parent class to ensure all tests get the benefit of this fix.

Reply:
If it's no way to happened, raise an exception to make it obvious and fix it.

Reply:
BTW, it's not a LISA bug, some systems are noexec to be more secure by design. LISA perfers to use commands, instead of scripts. If the script is rewritten in native LISA, there is no such error.
LISA shouldn't mark a mount point exec blindly, it may miss some test points.

Reply:
> LISA shouldn't mark a mount point exec blindly, it may miss some test points.
I wasn't clear on what the bug is, I think.

[My assumption on what working path is for is is based on the comments about what the working path is for:](https://vscode.dev/github/microsoft/lisa/blob/main/lisa/node.py#L187)

My impression is that the working path should be a place for tests to place things. Tests should be able to read and write things in the working path. I think a lot of test authors would assume they will be able to execute things in there as well. I think LISA should ensure that the working path (wherever it ends up) is r/w/x. LISA might have to be flexible about using available space or maybe even adding space.

> BTW, it's not a LISA bug, some systems are noexec to be more secure by design. LISA perfers to use commands, instead of scripts. If the script is rewritten in native LISA, there is no such error.

I don't think it's fair to blame the test for not checking whether the partition returned by` get_working_path()` is mounted with noexec. There was no mention of this mount option in the code before this PR, even in comments. The DF and Mount tool ignore mount options, so even if I had known about noexec and wanted to check the mount options before, I couldn't have using the tools provided by LISA.

I'm not trying to blame or shame anyone; bugs are a part of software development. Maybe bug is too strong of a word? LISA not checking the mount options of a partition before placing a working path is more of an undocumented non-intuitive behavior. 

It's not strange we haven't run into it before: it's only present on 1/6000th of the images in the marketplace.